bundles/org.openhab.binding.matter/src/main/java/org/openhab/binding/matter/internal/bridge/devices/FanDevice.java

```diff
@@ -0,0 +1,414 @@
+                            stringItem.send(new StringType(mappedMode));
+                        } else if (item instanceof SwitchItem switchItem) {
+                            switchItem.send(mode > 0 ? OnOffType.ON : OnOffType.OFF);
+                        }
+                    } catch (FanModeMappingException e) {
+                        logger.debug("Could not convert {} to custom value", data);
+                    }
+                    break;
+                case FanControlCluster.ATTRIBUTE_PERCENT_SETTING:
+                    int level = ((Double) data).intValue();
+                    if (item instanceof GroupItem groupItem) {
+                        groupItem.send(new PercentType(level));
+                    } else {
+                        ((DimmerItem) item).send(new PercentType(level));
+                    }
+                    break;
+                case OnOffCluster.ATTRIBUTE_ON_OFF:
+                    if (item instanceof SwitchItem switchItem) {
+                        OnOffType onOff = OnOffType.from((Boolean) data);
+                        switchItem.send(onOff);
+                        lastOnOff = onOff;
+                    }
+                    break;
+                default:
+                    break;
+            }
+        } else {
+            // if there is not an item bound to a specific attribute, we need to handle updating other items and fake it
+            switch (attributeName) {
+                case FanControlCluster.ATTRIBUTE_PERCENT_SETTING: {
+                    int level = ((Double) data).intValue();
+                    // try and update the on/off state if set
+                    GenericItem genericItem = itemForAttribute(OnOffCluster.CLUSTER_PREFIX,
+                            OnOffCluster.ATTRIBUTE_ON_OFF);
+                    if (genericItem instanceof SwitchItem switchItem) {
+                        switchItem.send(level > 0 ? OnOffType.ON : OnOffType.OFF);
+                    }
+                    // try and update the fan mode if set
+                    genericItem = itemForAttribute(FanControlCluster.CLUSTER_PREFIX,
+                            FanControlCluster.ATTRIBUTE_FAN_MODE);
+                    try {
+                        String mappedMode = fanModeMapper
+                                .toCustomValue(level > 0 ? FanControlCluster.FanModeEnum.ON.value
+                                        : FanControlCluster.FanModeEnum.OFF.value);
+                        if (genericItem instanceof NumberItem numberItem) {
+                            numberItem.send(new DecimalType(mappedMode));
+                        } else if (genericItem instanceof StringItem stringItem) {
+                            stringItem.send(new StringType(mappedMode));
+                        } else if (genericItem instanceof SwitchItem switchItem) {
+                            switchItem.send(level > 0 ? OnOffType.ON : OnOffType.OFF);
```

Comment:
```suggestion
                            switchItem.send(OnOffType.from(mode > 0));
```

Reply:
Since mapped mode can be a user defined string and is not always ON , we have to use the level value when dealing with Switch items.